### PR TITLE
Release str 523 sync event rpcs

### PR DESCRIPTION
**Pull request by delbonis**
## Description

Dup of #400 into release branch.

### Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update
-   [ ] Refactor

## Checklist

-   [x] I have performed a self-review of my code.
-   [x] I have commented my code where necessary.
-   [x] I have updated the documentation if needed.
-   [x] My changes do not introduce new warnings.
-   [ ] I have added tests that prove my changes are effective or that my feature works.
-   [x] New and existing tests pass with my changes.

### CODE REVIEW

**Comment by delbonis** [comment]
oh crap checks are failing

**Comment by storopoli** [comment]
Maybe we need to backport #387 ?

**Comment by delbonis** [comment]
Fixed git crap

File: crates/state/src/batch.rs [hunk 1]
```diff
@@ -206,10 +220,13 @@ impl BatchInfo {
...
 /// TODO: This needs to be replaced with GenesisState if we prove each Checkpoint
 /// recursively. Using a BootstrapState is a temporary solution
-#[derive(Clone, Debug, PartialEq, Eq, BorshDeserialize, BorshSerialize, Arbitrary)]
+#[derive(
+    Clone, Debug, PartialEq, Eq, Arbitrary, BorshDeserialize, BorshSerialize, Deserialize, Serialize,
+)]
 pub struct BootstrapState {
     pub idx: u64,
     pub start_l1_height: u64,
+    // TODO is this a blkid?
```
Comment (storopoli) [inline crates/state/src/batch.rs:229]: I think so

**Review by storopoli — APPROVED** [review]
ACK 12df6aaf45b2f43dd3ab17a10b9e748ab9a0e874

**Pull request merged**